### ISSUE TITLE
Add prose linter to CI

### ISSUE THREAD

**Issue by rht** [projectmesa/mesa#1572]
**What's the problem this feature will solve?**

We currently use codespell, but we could have a more uniform documentation style with a prose linter.

**Describe the solution you'd like**

One possible solution is using [Vale](https://github.com/errata-ai/vale). But there are other [alternatives](https://github.com/errata-ai/vale#functionality) as well.

**Additional context**
See https://github.com/projectmesa/mesa-examples/pull/14#issuecomment-1364454346

**Comment by rht** [comment]
@EwoutH any opinion on those prose linter tools?

**Comment by manvendra5345** [comment]
Hi everyone!
I am Manvendra Singh Solanki from Cse Ai-Ml background and i am also interested in working in this project , I know python , java , HTML , CSS  . I can adapt very fast and have great learning capacity but these thing are common what i think i have different in me is that i want to learn from thoes who are better than me and catch up  to them and even beat them and always keep learning.

I also want to contribute and learn but i am new to this platform please guide me .

Email:- singhmanvendra2003@gmail.com

**Comment by EwoutH** [comment]
> @EwoutH any opinion on those prose linter tools?

Don't know it, pitch it to me :)

**Comment by fammu03** [comment]
Hey is this issue still open? I would like to try and work on it.

**Comment by rht** [comment]
Yes, you can work on it. Vale is the one I have found to be used by multiple projects.

**Comment by sanika-n** [comment]
Hi, Sanika here, I just created a pull request which hopefully addresses this issue.. I integrated vale into the CLI workflow and configured so that vale gets triggered anytime anyone pushes code or submits a  PR, this is my first time contributing  in mesa, so pls let me know if it is alright:)

**Comment by EwoutH** [comment]
Thanks for the PR. I looked into it a bit, and I think a prose linter seems useful, and Vale looks like it's the most commonly used one and actively developed. So I support adding it, if it doesn't add too much overhead to our processes.

So go ahead, I left some more technical feedback in the PR.

**Comment by sanika-n** [comment]
Thank you, will try to resolve them as soon as possible

**Comment by Edu92337** [comment]
Hey @sanika-n , @EwoutH , I would like to contribute to this issue. Could you assign it to me?

**Comment by Sahil-Chhoker** [comment]
@Edu92337, you can look at the implementation and discussions in PR #2568 before working on the issue.

**Comment by pragam-m25** [comment]
Hi, I'm new to Project Mesa and very interested in contributing. I'm preparing to apply for GSoC 2026.

This issue of setting up a prose linter seems like a great task. I see that `Vale` is a suggested solution, and I'd like to take this on.

Can you please assign this issue to me?

**Comment by vedantvakharia** [comment]
Hi! I’ve completed the majority of the work for this issue and am nearly ready to open the PR. I just have two quick questions before I finalize the configuration:

1. Terminology Enforcement - I’ve added a custom mesa.yml style file that enforces project-specific terminology (e.g., suggesting "Mesa" instead of "mesa", "NumPy" instead of "numpy"). Could you please share a list of other specific terms or project names you would like standardized? I can add them to the rule set immediately.
2. Workflow - I have currently set Vale to run in two places: locally via pre-commit hooks, and automatically in CI via GitHub Actions. Do you want to keep both, or would you prefer it only runs in CI to keep the local setup simpler for contributors?

Looking forward to your feedback!
